libsql-wal/src/io/mod.rs

```diff
@@ -0,0 +1,99 @@
+use std::io;
+use std::path::Path;
+use std::sync::Arc;
+
+use chrono::{DateTime, Utc};
+use uuid::Uuid;
+
+use self::file::FileExt;
+
+pub mod buf;
+pub mod file;
+
+pub trait Io: Send + Sync + 'static {
+    type File: FileExt;
+    type TempFile: FileExt;
+
+    fn create_dir_all(&self, path: &Path) -> io::Result<()>;
+    fn open(
+        &self,
+        create_new: bool,
+        read: bool,
+        write: bool,
+        path: &Path,
+    ) -> io::Result<Self::File>;
+
+    // todo: create an async counterpart
+    fn tempfile(&self) -> io::Result<Self::TempFile>;
+    fn now(&self) -> DateTime<Utc>;
+    fn uuid(&self) -> Uuid;
```

Comment:
uuid is part of the io system 🤣

Reply:
it's randomness

Reply:
randomness is an input

Reply:
actually I'll probably just expose a range 🤔